File: converter-heic.py
import sys, os
from pathlib import Path
from PIL import Image
try:
    import pillow_heif
    pillow_heif.register_heif_opener()
except ImportError:
    print("Erro: pillow-heif não instalado. Execute: python -m pip install pillow-heif")
    sys.exit(1)

if len(sys.argv) < 2:
    print("Uso: python converter-heic.py <pasta_entrada> [pasta_saida]")
    print("Ex:  python converter-heic.py Q:\\fotos Q:\\fotos-convertidas")
    sys.exit(1)

entrada = Path(sys.argv[1])
saida = Path(sys.argv[2]) if len(sys.argv) > 2 else entrada

if not entrada.exists():
    print(f"Pasta de entrada não encontrada: {entrada}")
    sys.exit(1)

saida.mkdir(parents=True, exist_ok=True)

arquivos = sorted([f for f in entrada.iterdir() if f.suffix.lower() in (".heic", ".heif")])
total = len(arquivos)
print(f"\nEncontrados {total} arquivo(s) HEIC/HEIF em: {entrada}\n")

ok = err = 0
for i, f in enumerate(arquivos, 1):
    dest = saida / f.with_suffix(".jpg").name
    progresso = f"[{i}/{total}]"
    try:
        img = Image.open(f).convert("RGB")
        img.save(dest, "JPEG", quality=85)
        kb = dest.stat().st_size / 1024
        print(f"{progresso} OK  {f.name} -> {dest.name} ({kb:.0f}KB)")
        ok += 1
    except Exception as e:
        print(f"{progresso} ERRO {f.name}: {e}")
        err += 1

print(f"\nConcluído! {ok} convertidos, {err} erros.")
print(f"Pasta de saída: {saida}")

File: converter-heic.js
import sharp from "sharp";
import { readdirSync, statSync, mkdirSync, existsSync } from "fs";
import { join, parse, resolve } from "path";

process.env.LIBHEIF_MAX_NUMBER_OF_IREF_REFERENCES = "200";
process.env.LIBHEIF_MAX_NUMBER_OF_ILOC_ITEMS = "200";

const args = process.argv.slice(2);
if (args.length < 1) {
  console.log("Uso: node converter-heic.js <pasta_entrada> [pasta_saida]");
  console.log("Ex:  node converter-heic.js C:\\fotos C:\\fotos-convertidas");
  process.exit(1);
}

const entrada = resolve(args[0]);
const saida = args[1] ? resolve(args[1]) : entrada;

if (!existsSync(entrada)) {
  console.error("Pasta de entrada não encontrada:", entrada);
  process.exit(1);
}
if (!existsSync(saida)) mkdirSync(saida, { recursive: true });

const arquivos = readdirSync(entrada).filter(f => /\.(heic|heif)$/i.test(f));
const total = arquivos.length;
console.log(`\nEncontrados ${total} arquivo(s) HEIC/HEIF em: ${entrada}\n`);

let ok = 0, err = 0;
for (let i = 0; i < total; i++) {
  const f = arquivos[i];
  const src = join(entrada, f);
  const nome = parse(f).name;
  const dest = join(saida, nome + ".jpg");
  const progresso = `[${i + 1}/${total}]`;
  try {
    const buf = await sharp(src, { failOn: "none" }).jpeg({ quality: 85 }).toBuffer();
    await sharp(buf).toFile(dest);
    const kb = (statSync(dest).size / 1024).toFixed(0);
    console.log(`${progresso} OK  ${f} → ${nome}.jpg (${kb}KB)`);
    ok++;
  } catch (e) {
    console.error(`${progresso} ERRO ${f}: ${e.message}`);
    err++;
  }
}

console.log(`\nConcluído! ${ok} convertidos, ${err} erros.`);
console.log(`Pasta de saída: ${saida}`);
